internal/provider/resource_tfe_team_notification_configuration.go

```diff
@@ -323,7 +358,18 @@ func (r *resourceTFETeamNotificationConfiguration) Create(ctx context.Context, r
 		tnc.Token = plan.Token.ValueString()
 	}
 
-	result, diags := modelFromTFETeamNotificationConfiguration(tnc)
+	result, diags := modelFromTFETeamNotificationConfiguration(tnc, isWriteOnly, plan.Token)
+	if diags != nil {
```

Comment:
At the top of the definition of `modelFromTFETeamNotificationConfiguration`, you can instantiate a `diags` so that you can return `diag.Diagnostics` and not `*diag.Diagnostics`:

```
var diags diag.Diagnostics
```

And then here (and after other similar calls), we can check for errors the same way we do everywhere else:
```suggestion
	if diags.HasError() {
```